src/ui/public/index_patterns/_mapper.js

```diff
@@ -76,6 +77,10 @@ export default function MapperService(Private, Promise, es, esAdmin, config, kbn
       .then(function (fields) {
         fieldCache.set(id, fields);
         return fieldCache.get(id);
+      })
+      .catch(err => {
+        if (!opts.silenceErrors) notify.error(err);
```

Comment:
Can we just notify on the error from the caller? Supporting a `silenceErrors` options feels outside of the natural scope of `getFieldsForIndexPattern()`. If you keep it, please consider renaming the option to `sendErrorsToNotifier` or something; IMO silent errors don't produce error notifications.